week4/solutions.js

```diff
@@ -0,0 +1,166 @@
+import * as R from "ramda";
+import Maybe from "../week1/maybe";
+/* #########################################################################
+  Task1: 
+    Input: "function" 
+    Output: "Function"
+############################################################################ */
+R.compose(
+  R.converge(R.concat, [
+    R.compose(
+      R.toUpper,
+      R.head
+    ),
+    R.tail
+  ])
+)("function");
+/* #########################################################################
+ Task2: 
+    Input: [{id: 1, score: 90}, {id: 2, score: 40}, {id: 3, score: 80}]
+    Output: [{id: 1, score: 90}, {id: 2, score: 50}, {id: 3, score: 80}]
+    given id needs to be matched with 2 and score needs to be increased by 10
+############################################################################ */
+const task2_data = [
+  { id: 1, score: 90 },
+  { id: 2, score: 40 },
+  { id: 3, score: 80 }
+];
+R.adjust(
+  R.evolve({
+    score: R.add(10)
+  }),
+  R.findIndex(R.propEq("id", 2))(task2_data)
+)(task2_data);
+
+/* #########################################################################
+ Task3:
+     input: [1, 2, 3] 
+     output: [<div key={0}>1</div>, <div key={1}>2</div>, <div key={2}>3</div>]
+############################################################################ */
+var mapIndexed = R.addIndex(R.map);
+mapIndexed((val, idx) => "<div key=" + idx + ">" + idx + "</div>", [1, 2, 3]);
+
+/* #########################################################################
+ Task4: 
+     inpput: {name: "John", age: 20} 
+     output: [<div key="name">John</div>, <div key="age">20</div>]
+############################################################################ */
+var transforma = (num, key, obj) => "<div key=" + "key" + ">" + num + "</div>";
+R.values(R.mapObjIndexed(transforma, { name: "John", age: 20 }));
```

Comment:
Writing like the following as a habit so that you always create a function that can be composed later on.
```js
const format = compose(
  R.values,
  R.mapObjIndexed(transforma)
)
format({ name: "John", age: 20 })
```

Reply:
:+1: